### PR TITLE
WIP: Feature/presto view

### PR DESCRIPTION
**Pull request by arunravimv**

### CODE REVIEW

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    )
+    Preconditions.checkArgument(
+      data.endsWith(VIEW_SUFFIX),
+      "View data missing suffix: %s",
+      data
+    )
+
+    var encodedSQL = data.substring(VIEW_PREFIX.length)
+    encodedSQL = encodedSQL.substring(0, encodedSQL.length - VIEW_SUFFIX.length)
+    new String(Base64.getDecoder.decode(encodedSQL), StandardCharsets.UTF_8)
```
Comment (cvaliente) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala:62]: I think there is probably a more scala style of doing this

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    result
+  }
+
+  def isViewMarker(sqlText: String): Boolean = {
+    sqlText.startsWith(VIEW_PREFIX_MARKER)
+  }
+
+  def decode(sqlText: String): (String, StructType) = {
+    val result = decodeViewData(sqlText)
+    val prestoViewJson = GSON.fromJson(result, classOf[DecodedPrestoViewObject])
```
Comment (cvaliente) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala:37]: some debug logging of the extracted view here maybe?

Comment (arunravimv) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala:37]: Sure will add proper logs while doing the actual fix. This PR was just to share high-level logic.

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala (hunk 1 above)
Comment (arunravimv) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/prestoviews/PrestoViewDecoder.scala:62]: I will correct them when I make actual changes 👍

**Pull request closed without merging**